solsys_code/solsys_code_observatory/views.py

```diff
@@ -0,0 +1,151 @@
+from datetime import datetime, timezone
+from typing import Any
+
+import requests
+from django.contrib import messages
+from django.db.utils import IntegrityError
+from django.forms import BaseModelForm
+from django.http import HttpResponse
+from django.shortcuts import redirect
+from django.urls import reverse_lazy
+from django.views.generic import CreateView, DetailView, ListView
+from tom_catalogs.harvester import MissingDataException
+
+from solsys_code.solsys_code_observatory.forms import CreateObservatoryForm
+from solsys_code.solsys_code_observatory.models import Observatory
+
+
+class MPCObscodeFetcher:
+    """
+    The ``MPCObscodeFetcher`` is the interface to the Minor Planet Center (MPC)
+    Observatory Codes API
+    (https://www.minorplanetcenter.net/mpcops/documentation/obscodes-api/)
+    """
+
+    def query(self, obscode: str, dbg: bool = False):
+        """Query the MPC obscodes API for the specific <obscode>.
+        If successful, the JSON response data is stored in self.obs_data.
+
+        :param obscode: 3 character MPC observatory code to search for
+        :type term: str
+        :param dbg: Turns on basic print dump of the key-value pairs (or error response)
+        :type term: bool
+        """
+        self.obs_data = None
+
+        response = requests.get('https://data.minorplanetcenter.net/api/obscodes', json={'obscode': obscode})
+
+        if response.ok:
+            self.obs_data = response.json()
+            for key, value in response.json().items():
+                if dbg:
+                    print(f'{key:<27}: {value}')
```

Comment:
These should be logged rather than printed.

Reply:
Fixed in 366c28d